README.md
<p align="center"><img src="https://user-images.githubusercontent.com/56656996/99621835-b2d8dc80-29dd-11eb-8183-987e80f8b3a7.png"></p>
<h1 align="center">DDE Store</h1>
<h3 align="center">An app store for DDE built with DTK</h3>
<p align="center"><a href="/translations/README/README.es.md">🇪🇸 Español</a> | <a href="/translations/README/README.de.md">🇩🇪 Deutsch</a> | <a href="/translations/README/README.nl.md">🇳🇱 Nederlands</a> | <a href="/translations/README/README.ja.md">🇯🇵 日本語</a> | <a href="/translations/README/README.kab.md">🇩🇿 Taqbaylit</a> | <a href="/translations/README/README.tr.md">🇹🇷 Türkçe</a> | <a href="/translations/README/README.ar.md">🇸🇦 العربية</a></p>

## What is this?

This app store is designed for use with the Deepin Desktop Environment on other distributions. The official Deepin store is built around Debian and a web view with the Deepin repos on it, making it incompatible with other distros. This store is built with PackageKit and AppStream, so it should work perfect on [almost](#deepin-distro) any distro!

## Deepin Distro

This only applies to the official Deepin distro, not DDE in general. Deepin does not support AppStream, which provides this app with all the needed data. **This store is not supported on the Deepin distribution.** This was intended for other distros running DDE anyways, so I have no plans on ever supporting it. If Deepin starts using AppStream, I may consider it. If you're running Deepin and looking for an alternative app store, check out some other awesome projects like [Spark Store](https://www.spark-app.store) or [Deepines Store](https://deepines.com)!

## Installation
### UbuntuDDE
```
sudo apt-get install dde-store
```

### Arch
```
sudo pacman -S deepin-store
```

### Building from source
**Dependencies (Ubuntu)**
- cmake
- qtbase5-dev
- qttools5-dev
- libdtkwidget-dev
- libdtkcore-dev
- libdtkgui-dev
- libpackagekitqt5-dev
- libappstreamqt-dev
- libsnapd-qt-dev

**Building**
```
git clone https://github.com/dekzi/dde-store.git
cd dde-store
cmake .
sudo make install
```

## Contributions
Pull requests are always welcome for bug fixes, improvements, translations, or anything else!

## License
[GPLv3](LICENSE)


debian/changelog
dde-store (1.2.3) groovy; urgency=low

  * Added Japanese translation (by @MMP0)
  * Added Dutch translation (by @dutchdevil83)
  * Added Russian translation (by @thellage)
  * Added Kabyle translation (by @SelyanKab)
  * Bug fixes

 -- dekzi <dekzidev@protonmail.com>  Sun, 26 Jul 2021 00:47:12 -0800

dde-store (1.2.2) groovy; urgency=medium

  * Added "appstream://" and "snap://" url handling
  * Bug fixes

 -- dekzi <dekzidev@protonmail.com>  Mon, 22 Feb 2021 23:42:48 -0800

dde-store (1.2.1) groovy; urgency=low

  * Added Chinese translation (by @ShuiHuo)
  * Improved integration with additional backends (Snap)
  * Bug fixes

 -- dekzi <dekzidev@protonmail.com>  Sun, 17 Jan 2021 13:58:09 -0800

dde-store (1.2.0) groovy; urgency=low

  * Added Snap support
  * Bug fixes

 -- dekzi <dekzidev@protonmail.com>  Tue, 29 Dec 2020 22:32:20 -0800

dde-store (1.1.0) groovy; urgency=low

  * Added automatic hourly background update checking
  * Various performance improvements
  * Translation improvements
  * Bug fixes

 -- dekzi <dekzidev@protonmail.com>  Sun, 13 Dec 2020 16:52:52 -0800

dde-store (1.0.4) groovy; urgency=medium

  * Bug fixes

 -- dekzi <dekzidev@protonmail.com>  Mon, 07 Dec 2020 21:36:08 -0800

dde-store (1.0.3) groovy; urgency=low

  * Added Arabic translation (by @AhmadDakhlallah)
  * Bug fixes

 -- dekzi <dekzidev@protonmail.com>  Mon, 30 Nov 2020 17:37:49 -0800

dde-store (1.0.2) groovy; urgency=medium

  * Added Turkish translation (by @candancelik)
  * Bug fixes

 -- dekzi <dekzidev@protonmail.com>  Thu, 26 Nov 2020 23:07:38 -0800

dde-store (1.0.1) groovy; urgency=low

  * Added Spanish translation (by @Eboon-05)
  * Added German translation (by @Nesewebel)
  * Various bug fixes

 -- dekzi <dekzidev@protonmail.com>  Wed, 25 Nov 2020 12:17:49 -0800

dde-store (1.0.0) groovy; urgency=low

  * Initial release

 -- dekzi <dekzidev@protonmail.com>  Sun, 22 Nov 2020 16:17:00 -0800


dde-store_tr.ts
<?xml version="1.0" ?><!DOCTYPE TS><TS language="tr" version="2.1">
    <context>
        <name>main</name>
        <message>
            <location filename="../main.cpp" line="17"/>
            <source>An app store for DDE built with DTK</source>
            <translation>DTK ile oluşturulmuş DDE için bir uygulama mağazası</translation>
        </message>
        <message>
            <location filename="../main.cpp" line="17"/>
            <source>Created by %1</source>
            <translation>%1 tarafından oluşturuldu</translation>
        </message>
    </context>
    <context>
        <name>MainWindow</name>
        <message>
            <location filename="../mainwindow.cpp" line="91"/>
            <source>Settings</source>
            <translation>Ayarlar</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="115"/>
            <source>Home</source>
            <translation>Ana sayfa</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="116"/>
            <source>Messaging</source>
            <translation>Mesajlaşma</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="117"/>
            <source>Internet</source>
            <translation>İnternet</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="118"/>
            <source>Games</source>
            <translation>Oyunlar</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="119"/>
            <source>Development</source>
            <translation>Geliştirme</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="120"/>
            <source>Office</source>
            <translation>Ofis</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="121"/>
            <source>Graphics</source>
            <translation>Grafikler</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="122"/>
            <source>Video</source>
            <translation>Video</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="123"/>
            <source>Music</source>
            <translation>Müzik</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="124"/>
            <source>System</source>
            <translation>Sistem</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="125"/>
            <source>Installed</source>
            <translation>Yüklenenler</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="230"/>
            <source>Cannot close while app is being installed</source>
            <translation>Uygulama yüklenirken kapatılamaz</translation>
        </message>
        <message>
            <location filename="../mainwindow.cpp" line="231"/>
            <source>OK</source>
            <translation>Tamam</translation>
        </message>
    </context>
    <context>
        <name>CategoryPage</name>
        <message>
            <location filename="../pages/categorypage.cpp" line="15"/>
            <source>Sort by</source>
            <translation>Sıralama türü</translation>
        </message>
        <message>
            <location filename="../pages/categorypage.cpp" line="18"/>
            <source>Alphabetical</source>
            <translation>Alfabetık</translation>
        </message>
        <message>
            <location filename="../pages/categorypage.cpp" line="19"/>
            <source>Ratings</source>
            <translation>Derecelendirmeye göre</translation>
        </message>
        <message>
            <location filename="../pages/categorypage.cpp" line="39"/>
            <source>No results for %1</source>
            <translation>%1 için sonuç bulunamadı</translation>
        </message>
    </context>
    <context>
        <name>HomePage</name>
        <message>
            <location filename="../pages/homepage.cpp" line="50"/>
            <source>Messaging</source>
            <translation>Mesajlaşma</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="51"/>
            <source>Internet</source>
            <translation>İnternet</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="52"/>
            <source>Games</source>
            <translation>Oyunlar</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="53"/>
            <source>Development</source>
            <translation>Geliştirme</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="54"/>
            <source>Office</source>
            <translation>Ofis</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="55"/>
            <source>Graphics</source>
            <translation>Grafikler</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="56"/>
            <source>Video</source>
            <translation>Video</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="57"/>
            <source>Music</source>
            <translation>Müzik</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="58"/>
            <source>System</source>
            <translation>Sistem</translation>
        </message>
        <message>
            <location filename="../pages/homepage.cpp" line="91"/>
            <source>View More</source>
            <translation>Daha fazla</translation>
        </message>
    </context>
    <context>
        <name>ItemPage</name>
        <message>
            <location filename="../pages/itempage.cpp" line="62"/>
            <source>Uninstall</source>
            <translation>Kaldır</translation>
        </message>
        <message>
            <location filename="../pages/itempage.cpp" line="86"/>
            <source>No description provided.</source>
            <translation>Açıklama sağlanmadı.</translation>
        </message>
        <message>
            <location filename="../pages/itempage.cpp" line="101"/>
            <source>Install (%1)</source>
            <translation>Kur (%1)</translation>
        </message>
        <message>
            <location filename="../pages/itempage.cpp" line="110"/>
            <source>Open</source>
            <translation>Aç</translation>
        </message>
    </context>
    <context>
        <name>UpdatesPage</name>
        <message>
            <location filename="../pages/updatespage.cpp" line="12"/>
            <source>Updates</source>
            <translation>Güncellemeler</translation>
        </message>
        <message>
            <location filename="../pages/updatespage.cpp" line="13"/>
            <source>Up to date</source>
            <translation>Güncel</translation>
        </message>
        <message>
            <location filename="../pages/updatespage.cpp" line="15"/>
            <source>System Updates</source>
            <translation>Sistem güncellemeleri</translation>
        </message>
        <message>
            <location filename="../pages/updatespage.cpp" line="27"/>
            <source>Update All</source>
            <translation>Tümünü güncelle</translation>
        </message>
        <message>
            <location filename="../pages/updatespage.cpp" line="50"/>
            <source>Updating...</source>
            <translation>Güncelleniyor...</translation>
        </message>
        <message>
            <location filename="../pages/updatespage.cpp" line="53"/>
            <source>Update All (%1)</source>
            <translation>Tümünü güncelle (%1)</translation>
        </message>
    </context>
    <context>
        <name>stars</name>
        <message>
            <location filename="../widgets/stars.cpp" line="23"/>
            <source>(%1 ratings)</source>
            <translation>(%1 derecelendirme)</translation>
        </message>
    </context>
    <context>
        <name>AppStreamHelper</name>
        <message>
            <location filename="../backend/appstreamhelper.cpp" line="28"/>
            <source>ERROR: Unable to open AppStream metadata pool</source>
            <translation>Hata: AppStream metaveri havuzu açılamıyor</translation>
        </message>
        <message>
            <location filename="../backend/appstreamhelper.cpp" line="32"/>
            <source>OK</source>
            <translation>Tamam</translation>
        </message>
    </context>
    <context>
        <name>PackageKitHelper</name>
        <message>
            <location filename="../backend/packagekithelper.cpp" line="59"/>
            <source>Updates Available</source>
            <translation>Güncellemeler mevcut</translation>
        </message>
        <message>
            <location filename="../backend/packagekithelper.cpp" line="120"/>
            <source>Installed \"%1\"</source>
            <translation>Kuruldu \"%1\"</translation>
        </message>
        <message>
            <location filename="../backend/packagekithelper.cpp" line="140"/>
            <source>Uninstalled \"%1\"</source>
            <translation>Kaldırıldı \"%1\"</translation>
        </message>
        <message>
            <location filename="../backend/packagekithelper.cpp" line="167"/>
            <source>Updates Installed</source>
            <translation>Güncelleştirmeler kuruldu</translation>
        </message>
        <message>
            <location filename="../backend/packagekithelper.cpp" line="176"/>
            <source>PackageKit Error</source>
            <translation>PackageKit hatası</translation>
        </message>
    </context>
</TS>
